// nepali-calender-react/dist/index.modern.js
import React, { useState } from 'react';

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var b="function"===typeof Symbol&&Symbol.for,c=b?Symbol.for("react.element"):60103,d=b?Symbol.for("react.portal"):60106,e=b?Symbol.for("react.fragment"):60107,f=b?Symbol.for("react.strict_mode"):60108,g=b?Symbol.for("react.profiler"):60114,h=b?Symbol.for("react.provider"):60109,k=b?Symbol.for("react.context"):60110,l=b?Symbol.for("react.async_mode"):60111,m=b?Symbol.for("react.concurrent_mode"):60111,n=b?Symbol.for("react.forward_ref"):60112,p=b?Symbol.for("react.suspense"):60113,q=b?
Symbol.for("react.suspense_list"):60120,r=b?Symbol.for("react.memo"):60115,t=b?Symbol.for("react.lazy"):60116,v=b?Symbol.for("react.block"):60121,w=b?Symbol.for("react.fundamental"):60117,x=b?Symbol.for("react.responder"):60118,y=b?Symbol.for("react.scope"):60119;
function z(a){if("object"===typeof a&&null!==a){var u=a.$$typeof;switch(u){case c:switch(a=a.type,a){case l:case m:case e:case g:case f:case p:return a;default:switch(a=a&&a.$$typeof,a){case k:case n:case t:case r:case h:return a;default:return u}}case d:return u}}}function A(a){return z(a)===m}var AsyncMode=l;var ConcurrentMode=m;var ContextConsumer=k;var ContextProvider=h;var Element=c;var ForwardRef=n;var Fragment=e;var Lazy=t;var Memo=r;var Portal=d;
var Profiler=g;var StrictMode=f;var Suspense=p;var isAsyncMode=function(a){return A(a)||z(a)===l};var isConcurrentMode=A;var isContextConsumer=function(a){return z(a)===k};var isContextProvider=function(a){return z(a)===h};var isElement=function(a){return "object"===typeof a&&null!==a&&a.$$typeof===c};var isForwardRef=function(a){return z(a)===n};var isFragment=function(a){return z(a)===e};var isLazy=function(a){return z(a)===t};
var isMemo=function(a){return z(a)===r};var isPortal=function(a){return z(a)===d};var isProfiler=function(a){return z(a)===g};var isStrictMode=function(a){return z(a)===f};var isSuspense=function(a){return z(a)===p};
var isValidElementType=function(a){return "string"===typeof a||"function"===typeof a||a===e||a===m||a===g||a===f||a===p||a===q||"object"===typeof a&&null!==a&&(a.$$typeof===t||a.$$typeof===r||a.$$typeof===h||a.$$typeof===k||a.$$typeof===n||a.$$typeof===w||a.$$typeof===x||a.$$typeof===y||a.$$typeof===v)};var typeOf=z;

var reactIs_production_min = {
	AsyncMode: AsyncMode,
	ConcurrentMode: ConcurrentMode,
	ContextConsumer: ContextConsumer,
	ContextProvider: ContextProvider,
	Element: Element,
	ForwardRef: ForwardRef,
	Fragment: Fragment,
	Lazy: Lazy,
	Memo: Memo,
	Portal: Portal,
	Profiler: Profiler,
	StrictMode: StrictMode,
	Suspense: Suspense,
	isAsyncMode: isAsyncMode,
	isConcurrentMode: isConcurrentMode,
	isContextConsumer: isContextConsumer,
	isContextProvider: isContextProvider,
	isElement: isElement,
	isForwardRef: isForwardRef,
	isFragment: isFragment,
	isLazy: isLazy,
	isMemo: isMemo,
	isPortal: isPortal,
	isProfiler: isProfiler,
	isStrictMode: isStrictMode,
	isSuspense: isSuspense,
	isValidElementType: isValidElementType,
	typeOf: typeOf
};

var reactIs_development = createCommonjsModule(function (module, exports) {



if (process.env.NODE_ENV !== "production") {
  (function() {

// The Symbol used to tag the ReactElement-like types. If there is no native Symbol
// nor polyfill, then a plain number is used for performance.
var hasSymbol = typeof Symbol === 'function' && Symbol.for;
var REACT_ELEMENT_TYPE = hasSymbol ? Symbol.for('react.element') : 0xeac7;
var REACT_PORTAL_TYPE = hasSymbol ? Symbol.for('react.portal') : 0xeaca;
var REACT_FRAGMENT_TYPE = hasSymbol ? Symbol.for('react.fragment') : 0xeacb;
var REACT_STRICT_MODE_TYPE = hasSymbol ? Symbol.for('react.strict_mode') : 0xeacc;
var REACT_PROFILER_TYPE = hasSymbol ? Symbol.for('react.profiler') : 0xead2;
var REACT_PROVIDER_TYPE = hasSymbol ? Symbol.for('react.provider') : 0xeacd;
var REACT_CONTEXT_TYPE = hasSymbol ? Symbol.for('react.context') : 0xeace; // TODO: We don't use AsyncMode or ConcurrentMode anymore. They were temporary
// (unstable) APIs that have been removed. Can we remove the symbols?

var REACT_ASYNC_MODE_TYPE = hasSymbol ? Symbol.for('react.async_mode') : 0xeacf;
var REACT_CONCURRENT_MODE_TYPE = hasSymbol ? Symbol.for('react.concurrent_mode') : 0xeacf;
var REACT_FORWARD_REF_TYPE = hasSymbol ? Symbol.for('react.forward_ref') : 0xead0;
var REACT_SUSPENSE_TYPE = hasSymbol ? Symbol.for('react.suspense') : 0xead1;
var REACT_SUSPENSE_LIST_TYPE = hasSymbol ? Symbol.for('react.suspense_list') : 0xead8;
var REACT_MEMO_TYPE = hasSymbol ? Symbol.for('react.memo') : 0xead3;
var REACT_LAZY_TYPE = hasSymbol ? Symbol.for('react.lazy') : 0xead4;
var REACT_BLOCK_TYPE = hasSymbol ? Symbol.for('react.block') : 0xead9;
var REACT_FUNDAMENTAL_TYPE = hasSymbol ? Symbol.for('react.fundamental') : 0xead5;
var REACT_RESPONDER_TYPE = hasSymbol ? Symbol.for('react.responder') : 0xead6;
var REACT_SCOPE_TYPE = hasSymbol ? Symbol.for('react.scope') : 0xead7;

function isValidElementType(type) {
  return typeof type === 'string' || typeof type === 'function' || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
  type === REACT_FRAGMENT_TYPE || type === REACT_CONCURRENT_MODE_TYPE || type === REACT_PROFILER_TYPE || type === REACT_STRICT_MODE_TYPE || type === REACT_SUSPENSE_TYPE || type === REACT_SUSPENSE_LIST_TYPE || typeof type === 'object' && type !== null && (type.$$typeof === REACT_LAZY_TYPE || type.$$typeof === REACT_MEMO_TYPE || type.$$typeof === REACT_PROVIDER_TYPE || type.$$typeof === REACT_CONTEXT_TYPE || type.$$typeof === REACT_FORWARD_REF_TYPE || type.$$typeof === REACT_FUNDAMENTAL_TYPE || type.$$typeof === REACT_RESPONDER_TYPE || type.$$typeof === REACT_SCOPE_TYPE || type.$$typeof === REACT_BLOCK_TYPE);
}

function typeOf(object) {
  if (typeof object === 'object' && object !== null) {
    var $$typeof = object.$$typeof;

    switch ($$typeof) {
      case REACT_ELEMENT_TYPE:
        var type = object.type;

        switch (type) {
          case REACT_ASYNC_MODE_TYPE:
          case REACT_CONCURRENT_MODE_TYPE:
          case REACT_FRAGMENT_TYPE:
          case REACT_PROFILER_TYPE:
          case REACT_STRICT_MODE_TYPE:
          case REACT_SUSPENSE_TYPE:
            return type;

          default:
            var $$typeofType = type && type.$$typeof;

            switch ($$typeofType) {
              case REACT_CONTEXT_TYPE:
              case REACT_FORWARD_REF_TYPE:
              case REACT_LAZY_TYPE:
              case REACT_MEMO_TYPE:
              case REACT_PROVIDER_TYPE:
                return $$typeofType;

              default:
                return $$typeof;
            }

        }

      case REACT_PORTAL_TYPE:
        return $$typeof;
    }
  }

  return undefined;
} // AsyncMode is deprecated along with isAsyncMode

var AsyncMode = REACT_ASYNC_MODE_TYPE;
var ConcurrentMode = REACT_CONCURRENT_MODE_TYPE;
var ContextConsumer = REACT_CONTEXT_TYPE;
var ContextProvider = REACT_PROVIDER_TYPE;
var Element = REACT_ELEMENT_TYPE;
var ForwardRef = REACT_FORWARD_REF_TYPE;
var Fragment = REACT_FRAGMENT_TYPE;
var Lazy = REACT_LAZY_TYPE;
var Memo = REACT_MEMO_TYPE;
var Portal = REACT_PORTAL_TYPE;
var Profiler = REACT_PROFILER_TYPE;
var StrictMode = REACT_STRICT_MODE_TYPE;
var Suspense = REACT_SUSPENSE_TYPE;
var hasWarnedAboutDeprecatedIsAsyncMode = false; // AsyncMode should be deprecated

function isAsyncMode(object) {
  {
    if (!hasWarnedAboutDeprecatedIsAsyncMode) {
      hasWarnedAboutDeprecatedIsAsyncMode = true; // Using console['warn'] to evade Babel and ESLint

      console['warn']('The ReactIs.isAsyncMode() alias has been deprecated, ' + 'and will be removed in React 17+. Update your code to use ' + 'ReactIs.isConcurrentMode() instead. It has the exact same API.');
    }
  }

  return isConcurrentMode(object) || typeOf(object) === REACT_ASYNC_MODE_TYPE;
}
function isConcurrentMode(object) {
  return typeOf(object) === REACT_CONCURRENT_MODE_TYPE;
}
function isContextConsumer(object) {
  return typeOf(object) === REACT_CONTEXT_TYPE;
}
function isContextProvider(object) {
  return typeOf(object) === REACT_PROVIDER_TYPE;
}
function isElement(object) {
  return typeof object === 'object' && object !== null && object.$$typeof === REACT_ELEMENT_TYPE;
}
function isForwardRef(object) {
  return typeOf(object) === REACT_FORWARD_REF_TYPE;
}
function isFragment(object) {
  return typeOf(object) === REACT_FRAGMENT_TYPE;
}
function isLazy(object) {
  return typeOf(object) === REACT_LAZY_TYPE;
}
function isMemo(object) {
  return typeOf(object) === REACT_MEMO_TYPE;
}
function isPortal(object) {
  return typeOf(object) === REACT_PORTAL_TYPE;
}
function isProfiler(object) {
  return typeOf(object) === REACT_PROFILER_TYPE;
}
function isStrictMode(object) {
  return typeOf(object) === REACT_STRICT_MODE_TYPE;
}
function isSuspense(object) {
  return typeOf(object) === REACT_SUSPENSE_TYPE;
}

exports.AsyncMode = AsyncMode;
exports.ConcurrentMode = ConcurrentMode;
exports.ContextConsumer = ContextConsumer;
exports.ContextProvider = ContextProvider;
exports.Element = Element;
exports.ForwardRef = ForwardRef;
exports.Fragment = Fragment;
exports.Lazy = Lazy;
exports.Memo = Memo;
exports.Portal = Portal;
exports.Profiler = Profiler;
exports.StrictMode = StrictMode;
exports.Suspense = Suspense;
exports.isAsyncMode = isAsyncMode;
exports.isConcurrentMode = isConcurrentMode;
exports.isContextConsumer = isContextConsumer;
exports.isContextProvider = isContextProvider;
exports.isElement = isElement;
exports.isForwardRef = isForwardRef;
exports.isFragment = isFragment;
exports.isLazy = isLazy;
exports.isMemo = isMemo;
exports.isPortal = isPortal;
exports.isProfiler = isProfiler;
exports.isStrictMode = isStrictMode;
exports.isSuspense = isSuspense;
exports.isValidElementType = isValidElementType;
exports.typeOf = typeOf;
  })();
}
});

var reactIs = createCommonjsModule(function (module) {

if (process.env.NODE_ENV === 'production') {
  module.exports = reactIs_production_min;
} else {
  module.exports = reactIs_development;
}
});

/*
object-assign
(c) Sindre Sorhus
@license MIT
*/
/* eslint-disable no-unused-vars */
var getOwnPropertySymbols = Object.getOwnPropertySymbols;
var hasOwnProperty = Object.prototype.hasOwnProperty;
var propIsEnumerable = Object.prototype.propertyIsEnumerable;

function toObject(val) {
	if (val === null || val === undefined) {
		throw new TypeError('Object.assign cannot be called with null or undefined');
	}

	return Object(val);
}

function shouldUseNative() {
	try {
		if (!Object.assign) {
			return false;
		}

		// Detect buggy property enumeration order in older V8 versions.

		// https://bugs.chromium.org/p/v8/issues/detail?id=4118
		var test1 = new String('abc');  // eslint-disable-line no-new-wrappers
		test1[5] = 'de';
		if (Object.getOwnPropertyNames(test1)[0] === '5') {
			return false;
		}

		// https://bugs.chromium.org/p/v8/issues/detail?id=3056
		var test2 = {};
		for (var i = 0; i < 10; i++) {
			test2['_' + String.fromCharCode(i)] = i;
		}
		var order2 = Object.getOwnPropertyNames(test2).map(function (n) {
			return test2[n];
		});
		if (order2.join('') !== '0123456789') {
			return false;
		}

		// https://bugs.chromium.org/p/v8/issues/detail?id=3056
		var test3 = {};
		'abcdefghijklmnopqrst'.split('').forEach(function (letter) {
			test3[letter] = letter;
		});
		if (Object.keys(Object.assign({}, test3)).join('') !==
				'abcdefghijklmnopqrst') {
			return false;
		}

		return true;
	} catch (err) {
		// We don't expect any of the above to throw, but better to be safe.
		return false;
	}
}

var objectAssign = shouldUseNative() ? Object.assign : function (target, source) {
	var from;
	var to = toObject(target);
	var symbols;

	for (var s = 1; s < arguments.length; s++) {
		from = Object(arguments[s]);

		for (var key in from) {
			if (hasOwnProperty.call(from, key)) {
				to[key] = from[key];
			}
		}

		if (getOwnPropertySymbols) {
			symbols = getOwnPropertySymbols(from);
			for (var i = 0; i < symbols.length; i++) {
				if (propIsEnumerable.call(from, symbols[i])) {
					to[symbols[i]] = from[symbols[i]];
				}
			}
		}
	}

	return to;
};

/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

var ReactPropTypesSecret = 'SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED';

var ReactPropTypesSecret_1 = ReactPropTypesSecret;

var printWarning = function() {};

if (process.env.NODE_ENV !== 'production') {
  var ReactPropTypesSecret$1 = ReactPropTypesSecret_1;
  var loggedTypeFailures = {};
  var has = Function.call.bind(Object.prototype.hasOwnProperty);

  printWarning = function(text) {
    var message = 'Warning: ' + text;
    if (typeof console !== 'undefined') {
      console.error(message);
    }
    try {
      // --- Welcome to debugging React ---
      // This error was thrown as a convenience so that you can use this stack
      // to find the callsite that caused this warning to fire.
      throw new Error(message);
    } catch (x) {}
  };
}

/**
 * Assert that the values match with the type specs.
 * Error messages are memorized and will only be shown once.
 *
 * @param {object} typeSpecs Map of name to a ReactPropType
 * @param {object} values Runtime values that need to be type-checked
 * @param {string} location e.g. "prop", "context", "child context"
 * @param {string} componentName Name of the component for error messages.
 * @param {?Function} getStack Returns the component stack.
 * @private
 */
function checkPropTypes(typeSpecs, values, location, componentName, getStack) {
  if (process.env.NODE_ENV !== 'production') {
    for (var typeSpecName in typeSpecs) {
      if (has(typeSpecs, typeSpecName)) {
        var error;
        // Prop type validation may throw. In case they do, we don't want to
        // fail the render phase where it didn't fail before. So we log it.
        // After these have been cleaned up, we'll let them throw.
        try {
          // This is intentionally an invariant that gets caught. It's the same
          // behavior as without this statement except with a better message.
          if (typeof typeSpecs[typeSpecName] !== 'function') {
            var err = Error(
              (componentName || 'React class') + ': ' + location + ' type `' + typeSpecName + '` is invalid; ' +
              'it must be a function, usually from the `prop-types` package, but received `' + typeof typeSpecs[typeSpecName] + '`.'
            );
            err.name = 'Invariant Violation';
            throw err;
          }
          error = typeSpecs[typeSpecName](values, typeSpecName, componentName, location, null, ReactPropTypesSecret$1);
        } catch (ex) {
          error = ex;
        }
        if (error && !(error instanceof Error)) {
          printWarning(
            (componentName || 'React class') + ': type specification of ' +
            location + ' `' + typeSpecName + '` is invalid; the type checker ' +
            'function must return `null` or an `Error` but returned a ' + typeof error + '. ' +
            'You may have forgotten to pass an argument to the type checker ' +
            'creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and ' +
            'shape all require an argument).'
          );
        }
        if (error instanceof Error && !(error.message in loggedTypeFailures)) {
          // Only monitor this failure once because there tends to be a lot of the
          // same error.
          loggedTypeFailures[error.message] = true;

          var stack = getStack ? getStack() : '';

          printWarning(
            'Failed ' + location + ' type: ' + error.message + (stack != null ? stack : '')
          );
        }
      }
    }
  }
}

/**
 * Resets warning cache when testing.
 *
 * @private
 */
checkPropTypes.resetWarningCache = function() {
  if (process.env.NODE_ENV !== 'production') {
    loggedTypeFailures = {};
  }
};

var checkPropTypes_1 = checkPropTypes;

var has$1 = Function.call.bind(Object.prototype.hasOwnProperty);
var printWarning$1 = function() {};

if (process.env.NODE_ENV !== 'production') {
  printWarning$1 = function(text) {
    var message = 'Warning: ' + text;
    if (typeof console !== 'undefined') {
      console.error(message);
    }
    try {
      // --- Welcome to debugging React ---
      // This error was thrown as a convenience so that you can use this stack
      // to find the callsite that caused this warning to fire.
      throw new Error(message);
    } catch (x) {}
  };
}

function emptyFunctionThatReturnsNull() {
  return null;
}

var factoryWithTypeCheckers = function(isValidElement, throwOnDirectAccess) {
  /* global Symbol */
  var ITERATOR_SYMBOL = typeof Symbol === 'function' && Symbol.iterator;
  var FAUX_ITERATOR_SYMBOL = '@@iterator'; // Before Symbol spec.

  /**
   * Returns the iterator method function contained on the iterable object.
   *
   * Be sure to invoke the function with the iterable as context:
   *
   *     var iteratorFn = getIteratorFn(myIterable);
   *     if (iteratorFn) {
   *       var iterator = iteratorFn.call(myIterable);
   *       ...
   *     }
   *
   * @param {?object} maybeIterable
   * @return {?function}
   */
  function getIteratorFn(maybeIterable) {
    var iteratorFn = maybeIterable && (ITERATOR_SYMBOL && maybeIterable[ITERATOR_SYMBOL] || maybeIterable[FAUX_ITERATOR_SYMBOL]);
    if (typeof iteratorFn === 'function') {
      return iteratorFn;
    }
  }

  /**
   * Collection of methods that allow declaration and validation of props that are
   * supplied to React components. Example usage:
   *
   *   var Props = require('ReactPropTypes');
   *   var MyArticle = React.createClass({
   *     propTypes: {
   *       // An optional string prop named "description".
   *       description: Props.string,
   *
   *       // A required enum prop named "category".
   *       category: Props.oneOf(['News','Photos']).isRequired,
   *
   *       // A prop named "dialog" that requires an instance of Dialog.
   *       dialog: Props.instanceOf(Dialog).isRequired
   *     },
   *     render: function() { ... }
   *   });
   *
   * A more formal specification of how these methods are used:
   *
   *   type := array|bool|func|object|number|string|oneOf([...])|instanceOf(...)
   *   decl := ReactPropTypes.{type}(.isRequired)?
   *
   * Each and every declaration produces a function with the same signature. This
   * allows the creation of custom validation functions. For example:
   *
   *  var MyLink = React.createClass({
   *    propTypes: {
   *      // An optional string or URI prop named "href".
   *      href: function(props, propName, componentName) {
   *        var propValue = props[propName];
   *        if (propValue != null && typeof propValue !== 'string' &&
   *            !(propValue instanceof URI)) {
   *          return new Error(
   *            'Expected a string or an URI for ' + propName + ' in ' +
   *            componentName
   *          );
   *        }
   *      }
   *    },
   *    render: function() {...}
   *  });
   *
   * @internal
   */

  var ANONYMOUS = '<<anonymous>>';

  // Important!
  // Keep this list in sync with production version in `./factoryWithThrowingShims.js`.
  var ReactPropTypes = {
    array: createPrimitiveTypeChecker('array'),
    bool: createPrimitiveTypeChecker('boolean'),
    func: createPrimitiveTypeChecker('function'),
    number: createPrimitiveTypeChecker('number'),
    object: createPrimitiveTypeChecker('object'),
    string: createPrimitiveTypeChecker('string'),
    symbol: createPrimitiveTypeChecker('symbol'),

    any: createAnyTypeChecker(),
    arrayOf: createArrayOfTypeChecker,
    element: createElementTypeChecker(),
    elementType: createElementTypeTypeChecker(),
    instanceOf: createInstanceTypeChecker,
    node: createNodeChecker(),
    objectOf: createObjectOfTypeChecker,
    oneOf: createEnumTypeChecker,
    oneOfType: createUnionTypeChecker,
    shape: createShapeTypeChecker,
    exact: createStrictShapeTypeChecker,
  };

  /**
   * inlined Object.is polyfill to avoid requiring consumers ship their own
   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is
   */
  /*eslint-disable no-self-compare*/
  function is(x, y) {
    // SameValue algorithm
    if (x === y) {
      // Steps 1-5, 7-10
      // Steps 6.b-6.e: +0 != -0
      return x !== 0 || 1 / x === 1 / y;
    } else {
      // Step 6.a: NaN == NaN
      return x !== x && y !== y;
    }
  }
  /*eslint-enable no-self-compare*/

  /**
   * We use an Error-like object for backward compatibility as people may call
   * PropTypes directly and inspect their output. However, we don't use real
   * Errors anymore. We don't inspect their stack anyway, and creating them
   * is prohibitively expensive if they are created too often, such as what
   * happens in oneOfType() for any type before the one that matched.
   */
  function PropTypeError(message) {
    this.message = message;
    this.stack = '';
  }
  // Make `instanceof Error` still work for returned errors.
  PropTypeError.prototype = Error.prototype;

  function createChainableTypeChecker(validate) {
    if (process.env.NODE_ENV !== 'production') {
      var manualPropTypeCallCache = {};
      var manualPropTypeWarningCount = 0;
    }
    function checkType(isRequired, props, propName, componentName, location, propFullName, secret) {
      componentName = componentName || ANONYMOUS;
      propFullName = propFullName || propName;

      if (secret !== ReactPropTypesSecret_1) {
        if (throwOnDirectAccess) {
          // New behavior only for users of `prop-types` package
          var err = new Error(
            'Calling PropTypes validators directly is not supported by the `prop-types` package. ' +
            'Use `PropTypes.checkPropTypes()` to call them. ' +
            'Read more at http://fb.me/use-check-prop-types'
          );
          err.name = 'Invariant Violation';
          throw err;
        } else if (process.env.NODE_ENV !== 'production' && typeof console !== 'undefined') {
          // Old behavior for people using React.PropTypes
          var cacheKey = componentName + ':' + propName;
          if (
            !manualPropTypeCallCache[cacheKey] &&
            // Avoid spamming the console because they are often not actionable except for lib authors
            manualPropTypeWarningCount < 3
          ) {
            printWarning$1(
              'You are manually calling a React.PropTypes validation ' +
              'function for the `' + propFullName + '` prop on `' + componentName  + '`. This is deprecated ' +
              'and will throw in the standalone `prop-types` package. ' +
              'You may be seeing this warning due to a third-party PropTypes ' +
              'library. See https://fb.me/react-warning-dont-call-proptypes ' + 'for details.'
            );
            manualPropTypeCallCache[cacheKey] = true;
            manualPropTypeWarningCount++;
          }
        }
      }
      if (props[propName] == null) {
        if (isRequired) {
          if (props[propName] === null) {
            return new PropTypeError('The ' + location + ' `' + propFullName + '` is marked as required ' + ('in `' + componentName + '`, but its value is `null`.'));
          }
          return new PropTypeError('The ' + location + ' `' + propFullName + '` is marked as required in ' + ('`' + componentName + '`, but its value is `undefined`.'));
        }
        return null;
      } else {
        return validate(props, propName, componentName, location, propFullName);
      }
    }

    var chainedCheckType = checkType.bind(null, false);
    chainedCheckType.isRequired = checkType.bind(null, true);

    return chainedCheckType;
  }

  function createPrimitiveTypeChecker(expectedType) {
    function validate(props, propName, componentName, location, propFullName, secret) {
      var propValue = props[propName];
      var propType = getPropType(propValue);
      if (propType !== expectedType) {
        // `propValue` being instance of, say, date/regexp, pass the 'object'
        // check, but we can offer a more precise error message here rather than
        // 'of type `object`'.
        var preciseType = getPreciseType(propValue);

        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + preciseType + '` supplied to `' + componentName + '`, expected ') + ('`' + expectedType + '`.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }

  function createAnyTypeChecker() {
    return createChainableTypeChecker(emptyFunctionThatReturnsNull);
  }

  function createArrayOfTypeChecker(typeChecker) {
    function validate(props, propName, componentName, location, propFullName) {
      if (typeof typeChecker !== 'function') {
        return new PropTypeError('Property `' + propFullName + '` of component `' + componentName + '` has invalid PropType notation inside arrayOf.');
      }
      var propValue = props[propName];
      if (!Array.isArray(propValue)) {
        var propType = getPropType(propValue);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected an array.'));
      }
      for (var i = 0; i < propValue.length; i++) {
        var error = typeChecker(propValue, i, componentName, location, propFullName + '[' + i + ']', ReactPropTypesSecret_1);
        if (error instanceof Error) {
          return error;
        }
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }

  function createElementTypeChecker() {
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      if (!isValidElement(propValue)) {
        var propType = getPropType(propValue);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected a single ReactElement.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }

  function createElementTypeTypeChecker() {
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      if (!reactIs.isValidElementType(propValue)) {
        var propType = getPropType(propValue);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected a single ReactElement type.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }

  function createInstanceTypeChecker(expectedClass) {
    function validate(props, propName, componentName, location, propFullName) {
      if (!(props[propName] instanceof expectedClass)) {
        var expectedClassName = expectedClass.name || ANONYMOUS;
        var actualClassName = getClassName(props[propName]);
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + actualClassName + '` supplied to `' + componentName + '`, expected ') + ('instance of `' + expectedClassName + '`.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }

  function createEnumTypeChecker(expectedValues) {
    if (!Array.isArray(expectedValues)) {
      if (process.env.NODE_ENV !== 'production') {
        if (arguments.length > 1) {
          printWarning$1(
            'Invalid arguments supplied to oneOf, expected an array, got ' + arguments.length + ' arguments. ' +
            'A common mistake is to write oneOf(x, y, z) instead of oneOf([x, y, z]).'
          );
        } else {
          printWarning$1('Invalid argument supplied to oneOf, expected an array.');
        }
      }
      return emptyFunctionThatReturnsNull;
    }

    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      for (var i = 0; i < expectedValues.length; i++) {
        if (is(propValue, expectedValues[i])) {
          return null;
        }
      }

      var valuesString = JSON.stringify(expectedValues, function replacer(key, value) {
        var type = getPreciseType(value);
        if (type === 'symbol') {
          return String(value);
        }
        return value;
      });
      return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of value `' + String(propValue) + '` ' + ('supplied to `' + componentName + '`, expected one of ' + valuesString + '.'));
    }
    return createChainableTypeChecker(validate);
  }

  function createObjectOfTypeChecker(typeChecker) {
    function validate(props, propName, componentName, location, propFullName) {
      if (typeof typeChecker !== 'function') {
        return new PropTypeError('Property `' + propFullName + '` of component `' + componentName + '` has invalid PropType notation inside objectOf.');
      }
      var propValue = props[propName];
      var propType = getPropType(propValue);
      if (propType !== 'object') {
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + ('`' + propType + '` supplied to `' + componentName + '`, expected an object.'));
      }
      for (var key in propValue) {
        if (has$1(propValue, key)) {
          var error = typeChecker(propValue, key, componentName, location, propFullName + '.' + key, ReactPropTypesSecret_1);
          if (error instanceof Error) {
            return error;
          }
        }
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }

  function createUnionTypeChecker(arrayOfTypeCheckers) {
    if (!Array.isArray(arrayOfTypeCheckers)) {
      process.env.NODE_ENV !== 'production' ? printWarning$1('Invalid argument supplied to oneOfType, expected an instance of array.') : void 0;
      return emptyFunctionThatReturnsNull;
    }

    for (var i = 0; i < arrayOfTypeCheckers.length; i++) {
      var checker = arrayOfTypeCheckers[i];
      if (typeof checker !== 'function') {
        printWarning$1(
          'Invalid argument supplied to oneOfType. Expected an array of check functions, but ' +
          'received ' + getPostfixForTypeWarning(checker) + ' at index ' + i + '.'
        );
        return emptyFunctionThatReturnsNull;
      }
    }

    function validate(props, propName, componentName, location, propFullName) {
      for (var i = 0; i < arrayOfTypeCheckers.length; i++) {
        var checker = arrayOfTypeCheckers[i];
        if (checker(props, propName, componentName, location, propFullName, ReactPropTypesSecret_1) == null) {
          return null;
        }
      }

      return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` supplied to ' + ('`' + componentName + '`.'));
    }
    return createChainableTypeChecker(validate);
  }

  function createNodeChecker() {
    function validate(props, propName, componentName, location, propFullName) {
      if (!isNode(props[propName])) {
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` supplied to ' + ('`' + componentName + '`, expected a ReactNode.'));
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }

  function createShapeTypeChecker(shapeTypes) {
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      var propType = getPropType(propValue);
      if (propType !== 'object') {
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type `' + propType + '` ' + ('supplied to `' + componentName + '`, expected `object`.'));
      }
      for (var key in shapeTypes) {
        var checker = shapeTypes[key];
        if (!checker) {
          continue;
        }
        var error = checker(propValue, key, componentName, location, propFullName + '.' + key, ReactPropTypesSecret_1);
        if (error) {
          return error;
        }
      }
      return null;
    }
    return createChainableTypeChecker(validate);
  }

  function createStrictShapeTypeChecker(shapeTypes) {
    function validate(props, propName, componentName, location, propFullName) {
      var propValue = props[propName];
      var propType = getPropType(propValue);
      if (propType !== 'object') {
        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type `' + propType + '` ' + ('supplied to `' + componentName + '`, expected `object`.'));
      }
      // We need to check all keys in case some are required but missing from
      // props.
      var allKeys = objectAssign({}, props[propName], shapeTypes);
      for (var key in allKeys) {
        var checker = shapeTypes[key];
        if (!checker) {
          return new PropTypeError(
            'Invalid ' + location + ' `' + propFullName + '` key `' + key + '` supplied to `' + componentName + '`.' +
            '\nBad object: ' + JSON.stringify(props[propName], null, '  ') +
            '\nValid keys: ' +  JSON.stringify(Object.keys(shapeTypes), null, '  ')
          );
        }
        var error = checker(propValue, key, componentName, location, propFullName + '.' + key, ReactPropTypesSecret_1);
        if (error) {
          return error;
        }
      }
      return null;
    }

    return createChainableTypeChecker(validate);
  }

  function isNode(propValue) {
    switch (typeof propValue) {
      case 'number':
      case 'string':
      case 'undefined':
        return true;
      case 'boolean':
        return !propValue;
      case 'object':
        if (Array.isArray(propValue)) {
          return propValue.every(isNode);
        }
        if (propValue === null || isValidElement(propValue)) {
          return true;
        }

        var iteratorFn = getIteratorFn(propValue);
        if (iteratorFn) {
          var iterator = iteratorFn.call(propValue);
          var step;
          if (iteratorFn !== propValue.entries) {
            while (!(step = iterator.next()).done) {
              if (!isNode(step.value)) {
                return false;
              }
            }
          } else {
            // Iterator will provide entry [k,v] tuples rather than values.
            while (!(step = iterator.next()).done) {
              var entry = step.value;
              if (entry) {
                if (!isNode(entry[1])) {
                  return false;
                }
              }
            }
          }
        } else {
          return false;
        }

        return true;
      default:
        return false;
    }
  }

  function isSymbol(propType, propValue) {
    // Native Symbol.
    if (propType === 'symbol') {
      return true;
    }

    // falsy value can't be a Symbol
    if (!propValue) {
      return false;
    }

    // 19.4.3.5 Symbol.prototype[@@toStringTag] === 'Symbol'
    if (propValue['@@toStringTag'] === 'Symbol') {
      return true;
    }

    // Fallback for non-spec compliant Symbols which are polyfilled.
    if (typeof Symbol === 'function' && propValue instanceof Symbol) {
      return true;
    }

    return false;
  }

  // Equivalent of `typeof` but with special handling for array and regexp.
  function getPropType(propValue) {
    var propType = typeof propValue;
    if (Array.isArray(propValue)) {
      return 'array';
    }
    if (propValue instanceof RegExp) {
      // Old webkits (at least until Android 4.0) return 'function' rather than
      // 'object' for typeof a RegExp. We'll normalize this here so that /bla/
      // passes PropTypes.object.
      return 'object';
    }
    if (isSymbol(propType, propValue)) {
      return 'symbol';
    }
    return propType;
  }

  // This handles more types than `getPropType`. Only used for error messages.
  // See `createPrimitiveTypeChecker`.
  function getPreciseType(propValue) {
    if (typeof propValue === 'undefined' || propValue === null) {
      return '' + propValue;
    }
    var propType = getPropType(propValue);
    if (propType === 'object') {
      if (propValue instanceof Date) {
        return 'date';
      } else if (propValue instanceof RegExp) {
        return 'regexp';
      }
    }
    return propType;
  }

  // Returns a string that is postfixed to a warning about an invalid type.
  // For example, "undefined" or "of type array"
  function getPostfixForTypeWarning(value) {
    var type = getPreciseType(value);
    switch (type) {
      case 'array':
      case 'object':
        return 'an ' + type;
      case 'boolean':
      case 'date':
      case 'regexp':
        return 'a ' + type;
      default:
        return type;
    }
  }

  // Returns class name of the object, if any.
  function getClassName(propValue) {
    if (!propValue.constructor || !propValue.constructor.name) {
      return ANONYMOUS;
    }
    return propValue.constructor.name;
  }

  ReactPropTypes.checkPropTypes = checkPropTypes_1;
  ReactPropTypes.resetWarningCache = checkPropTypes_1.resetWarningCache;
  ReactPropTypes.PropTypes = ReactPropTypes;

  return ReactPropTypes;
};

function emptyFunction() {}
function emptyFunctionWithReset() {}
emptyFunctionWithReset.resetWarningCache = emptyFunction;

var factoryWithThrowingShims = function() {
  function shim(props, propName, componentName, location, propFullName, secret) {
    if (secret === ReactPropTypesSecret_1) {
      // It is still safe when called from React.
      return;
    }
    var err = new Error(
      'Calling PropTypes validators directly is not supported by the `prop-types` package. ' +
      'Use PropTypes.checkPropTypes() to call them. ' +
      'Read more at http://fb.me/use-check-prop-types'
    );
    err.name = 'Invariant Violation';
    throw err;
  }  shim.isRequired = shim;
  function getShim() {
    return shim;
  }  // Important!
  // Keep this list in sync with production version in `./factoryWithTypeCheckers.js`.
  var ReactPropTypes = {
    array: shim,
    bool: shim,
    func: shim,
    number: shim,
    object: shim,
    string: shim,
    symbol: shim,

    any: shim,
    arrayOf: getShim,
    element: shim,
    elementType: shim,
    instanceOf: getShim,
    node: shim,
    objectOf: getShim,
    oneOf: getShim,
    oneOfType: getShim,
    shape: getShim,
    exact: getShim,

    checkPropTypes: emptyFunctionWithReset,
    resetWarningCache: emptyFunction
  };

  ReactPropTypes.PropTypes = ReactPropTypes;

  return ReactPropTypes;
};

var propTypes = createCommonjsModule(function (module) {
/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

if (process.env.NODE_ENV !== 'production') {
  var ReactIs = reactIs;

  // By explicitly using `prop-types` you are opting into new development behavior.
  // http://fb.me/prop-types-in-prod
  var throwOnDirectAccess = true;
  module.exports = factoryWithTypeCheckers(ReactIs.isElement, throwOnDirectAccess);
} else {
  // By explicitly using `prop-types` you are opting into new production behavior.
  // http://fb.me/prop-types-in-prod
  module.exports = factoryWithThrowingShims();
}
});

var styles = {"calendarCcontain":"_1D41V","calendarDays":"_gHhFr","calendarTopbar":"_3dwv1","topbarDays":"_30eBi","saturday":"_13aKv","calendarWeek":"_jxd3s","calendarDay":"_2BApa","active":"_3sodH","holiday":"_1Kx67","calendarDate":"_1KwS1","inactive":"_3hvs3","taskCount":"_AiVxd","today":"_B6tPl","calendarTask":"_2Z61O","calendarTaskToday":"_3mOVN"};

var bsMonths = ['बैशाख', 'जेठ', 'आषाढ', 'श्रावण', 'भदौ', 'असोज', 'कार्तिक', 'मंसिर', 'पौष', 'माघ', 'फाल्गुन', 'चैत्र'];
var bsMonthsEnglish = ['Baisakh', 'Jesth', 'Aasad', 'Srawan', 'Bhadra', 'Aaswin', 'Kartik', 'Mangsir', 'Paush', 'Magh', 'Falgun', 'Chaitra'];
var bsDays = ['आईत', 'सोम', 'मंगल', 'बुध', 'बिही', 'शुक्र', 'शनि'];
var bsDaysFull = ['आइतवार', 'सोमवार', 'मंगलवार', 'बुधवार', 'बिहिवार', 'शुक्रवार', 'शनिवार'];
var adDaysFull = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
var adDays = ['Sun', 'Mon', 'Tues', 'Wed', 'Thu', 'Fri', 'Sat'];
var nepaliNumbers = ['०', '१', '२', '३', '४', '५', '६', '७', '८', '९'];
var adMonths = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

var dates = {
  1970: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  1971: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  1972: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  1973: [{
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 31
  }],
  1974: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }],
  1975: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  1976: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 31
  }],
  1977: [{
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 31
  }],
  1978: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  1979: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  1980: [{
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 31
  }],
  1981: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  1982: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  1983: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }],
  1984: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 31
  }],
  1985: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }],
  1986: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  1987: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  1988: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 31
  }],
  1989: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  1990: [{
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  1991: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }],
  1992: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }],
  1993: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  1994: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }],
  1995: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  1996: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  1997: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  1998: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  1999: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  2000: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 31
  }],
  2001: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  2002: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }],
  2003: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 31
  }],
  2004: [{
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 31
  }],
  2005: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }],
  2006: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  2007: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 31
  }],
  2008: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 31
  }],
  2009: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  2010: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  2011: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 31
  }],
  2012: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  2013: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }],
  2014: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  2015: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 31
  }],
  2016: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }],
  2017: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  2018: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  2019: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 31
  }],
  2020: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  2021: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  2022: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }],
  2023: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 31
  }],
  2024: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }],
  2025: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  2026: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 31
  }],
  2027: [{
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 31
  }],
  2028: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  2029: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  2030: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 31
  }],
  2031: [{
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 31
  }],
  2032: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  2033: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }],
  2034: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 31
  }],
  2035: [{
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 31
  }],
  2036: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  2037: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  2038: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 31
  }],
  2039: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  2040: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  2041: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }],
  2042: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 31
  }],
  2043: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  2044: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }],
  2045: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  2046: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 31
  }],
  2047: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  2048: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  2049: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  2050: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 31
  }],
  2051: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  2052: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }],
  2053: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  2054: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 31
  }],
  2055: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }],
  2056: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  2057: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 31
  }],
  2058: [{
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 31
  }],
  2059: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  2060: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  2061: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 31
  }],
  2062: [{
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 31
  }],
  2063: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }],
  2064: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  2065: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 31
  }],
  2066: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 31
  }],
  2067: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  2068: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  2069: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 31
  }],
  2070: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  2071: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  2072: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }],
  2073: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 31
  }],
  2074: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }],
  2075: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  2076: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  2077: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 31
  }],
  2078: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  2079: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  2080: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }],
  2081: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  2082: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  2083: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }],
  2084: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  2085: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 29
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }],
  2086: [{
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  2087: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  2088: [{
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }],
  2089: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }],
  2090: [{
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  2091: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 32
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  2092: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }],
  2093: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  2094: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }],
  2095: [{
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 32
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 29
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  2096: [{
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 32
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 29
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }],
  2097: [{
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 32
  }, {
    weekStart: 5,
    endDate: 31
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 29
  }, {
    weekStart: 1,
    endDate: 30
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }],
  2098: [{
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 32
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 31
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 31
  }],
  2099: [{
    weekStart: 1,
    endDate: 31
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 32
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 31
  }, {
    weekStart: 3,
    endDate: 31
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 29
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 30
  }, {
    weekStart: 0,
    endDate: 30
  }],
  2100: [{
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 31
  }, {
    weekStart: 5,
    endDate: 32
  }, {
    weekStart: 2,
    endDate: 30
  }, {
    weekStart: 4,
    endDate: 31
  }, {
    weekStart: 0,
    endDate: 30
  }, {
    weekStart: 2,
    endDate: 29
  }, {
    weekStart: 3,
    endDate: 30
  }, {
    weekStart: 5,
    endDate: 29
  }, {
    weekStart: 6,
    endDate: 30
  }, {
    weekStart: 1,
    endDate: 30
  }]
};
var currentYear = function currentYear(year) {
  return dates[year];
};
var getYears = function getYears(year) {
  return Object.keys(dates);
};

var BikramSambatConverter = function BikramSambatConverter() {
  this.isLeapYear = function (year) {
    var a = year;

    if (a % 100 === 0) {
      if (a % 400 === 0) {
        return true;
      } else {
        return false;
      }
    } else {
      if (a % 4 === 0) {
        return true;
      } else {
        return false;
      }
    }
  };

  this.getNepaliMonth = function (m) {
    return this.nepaliMonths[m - 1];
  };

  this.getEnglishMonth = function (m) {
    return this.engMonths[m - 1];
  };

  this.getDayOfTheWeek = function (day) {
    return this.nepaliWeekdays[day - 1];
  };

  this.getEngDayOfTheWeek = function (day) {
    return this.engWeekDays[day - 1];
  };

  this.isRangeEng = function (yy, mm, dd) {
    if (yy < 1944 || yy > 2033) {
      this.debugInfo = 'Supported only between 1944-2022';
      return false;
    }

    if (mm < 1 || mm > 12) {
      this.debugInfo = 'Error! value 1-12 only';
      return false;
    }

    if (dd < 1 || dd > 31) {
      this.debugInfo = 'Error! value 1-31 only';
      return false;
    }

    return true;
  };

  this.toNepaliNumber = function (number) {
    var strNum = String(number);
    var strNepNum = '';

    for (var i = 0; i < strNum.length; i++) {
      strNepNum += this.nepDigits[parseInt(strNum.substr(i, 1))];
    }

    return strNepNum;
  };

  this.engToNepali = function (yy, mm, dd) {
    yy = parseInt(yy);
    mm = parseInt(mm);
    dd = parseInt(dd);

    if (this.isRangeEng(yy, mm, dd) === false) {
      return false;
    } else {
      var month = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
      var lmonth = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
      var defEyy = 1944;
      var defNyy = 2000;
      var defNum = 9;
      var defNdd = 17 - 1;
      var totalEdays = 0;
      var totalNdays = 0;
      var a = 0;
      var day = 7 - 1;
      var m = 0;
      var y = 0;
      var i = 0;
      var j = 0;
      var numDay = 0;

      for (i = 0; i < yy - defEyy; i++) {
        if (this.isLeapYear(defEyy + i)) for (j = 0; j < 12; j++) {
          totalEdays += lmonth[j];
        } else for (j = 0; j < 12; j++) {
          totalEdays += month[j];
        }
      }

      for (i = 0; i < mm - 1; i++) {
        if (this.isLeapYear(yy)) totalEdays += lmonth[i];else totalEdays += month[i];
      }

      totalEdays += dd;
      i = 0;
      j = defNum;
      totalNdays = defNdd;
      m = defNum;
      y = defNyy;

      while (totalEdays !== 0) {
        a = this.bs[i][j];
        totalNdays++;
        day++;

        if (totalNdays > a) {
          m++;
          totalNdays = 1;
          j++;
        }

        if (day > 7) day = 1;

        if (m > 12) {
          y++;
          m = 1;
        }

        if (j > 12) {
          j = 1;
          i++;
        }

        totalEdays--;
      }

      numDay = day;
      this.nepDate = {};
      this.nepDate.year = y;
      this.nepDate.month = this.getNepaliMonth(m);
      this.nepDate.date = totalNdays;
      this.nepDate.day = this.getEngDayOfTheWeek(day);
      this.nepDate.nmonth = m;
      this.nepDate.num_day = numDay;
      return this.nepDate;
    }
  };

  this.bs = {
    0: [2000, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 29, 31],
    1: [2001, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    2: [2002, 31, 31, 32, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    3: [2003, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    4: [2004, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 29, 31],
    5: [2005, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    6: [2006, 31, 31, 32, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    7: [2007, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    8: [2008, 31, 31, 31, 32, 31, 31, 29, 30, 30, 29, 29, 31],
    9: [2009, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    10: [2010, 31, 31, 32, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    11: [2011, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    12: [2012, 31, 31, 31, 32, 31, 31, 29, 30, 30, 29, 30, 30],
    13: [2013, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    14: [2014, 31, 31, 32, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    15: [2015, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    16: [2016, 31, 31, 31, 32, 31, 31, 29, 30, 30, 29, 30, 30],
    17: [2017, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    18: [2018, 31, 32, 31, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    19: [2019, 31, 32, 31, 32, 31, 30, 30, 30, 29, 30, 29, 31],
    20: [2020, 31, 31, 31, 32, 31, 31, 30, 29, 30, 29, 30, 30],
    21: [2021, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    22: [2022, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 30],
    23: [2023, 31, 32, 31, 32, 31, 30, 30, 30, 29, 30, 29, 31],
    24: [2024, 31, 31, 31, 32, 31, 31, 30, 29, 30, 29, 30, 30],
    25: [2025, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    26: [2026, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    27: [2027, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 29, 31],
    28: [2028, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    29: [2029, 31, 31, 32, 31, 32, 30, 30, 29, 30, 29, 30, 30],
    30: [2030, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    31: [2031, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 29, 31],
    32: [2032, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    33: [2033, 31, 31, 32, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    34: [2034, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    35: [2035, 30, 32, 31, 32, 31, 31, 29, 30, 30, 29, 29, 31],
    36: [2036, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    37: [2037, 31, 31, 32, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    38: [2038, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    39: [2039, 31, 31, 31, 32, 31, 31, 29, 30, 30, 29, 30, 30],
    40: [2040, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    41: [2041, 31, 31, 32, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    42: [2042, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    43: [2043, 31, 31, 31, 32, 31, 31, 29, 30, 30, 29, 30, 30],
    44: [2044, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    45: [2045, 31, 32, 31, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    46: [2046, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    47: [2047, 31, 31, 31, 32, 31, 31, 30, 29, 30, 29, 30, 30],
    48: [2048, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    49: [2049, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 30],
    50: [2050, 31, 32, 31, 32, 31, 30, 30, 30, 29, 30, 29, 31],
    51: [2051, 31, 31, 31, 32, 31, 31, 30, 29, 30, 29, 30, 30],
    52: [2052, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    53: [2053, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 30],
    54: [2054, 31, 32, 31, 32, 31, 30, 30, 30, 29, 30, 29, 31],
    55: [2055, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    56: [2056, 31, 31, 32, 31, 32, 30, 30, 29, 30, 29, 30, 30],
    57: [2057, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    58: [2058, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 29, 31],
    59: [2059, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    60: [2060, 31, 31, 32, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    61: [2061, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    62: [2062, 30, 32, 31, 32, 31, 31, 29, 30, 29, 30, 29, 31],
    63: [2063, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    64: [2064, 31, 31, 32, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    65: [2065, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    66: [2066, 31, 31, 31, 32, 31, 31, 29, 30, 30, 29, 29, 31],
    67: [2067, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    68: [2068, 31, 31, 32, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    69: [2069, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    70: [2070, 31, 31, 31, 32, 31, 31, 29, 30, 30, 29, 30, 30],
    71: [2071, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    72: [2072, 31, 32, 31, 32, 31, 30, 30, 29, 30, 29, 30, 30],
    73: [2073, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31],
    74: [2074, 31, 31, 31, 32, 31, 31, 30, 29, 30, 29, 30, 30],
    75: [2075, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    76: [2076, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 30],
    77: [2077, 31, 32, 31, 32, 31, 30, 30, 30, 29, 30, 29, 31],
    78: [2078, 31, 31, 31, 32, 31, 31, 30, 29, 30, 29, 30, 30],
    79: [2079, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30],
    80: [2080, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 30],
    81: [2081, 31, 31, 32, 32, 31, 30, 30, 30, 29, 30, 30, 30],
    82: [2082, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 30, 30],
    83: [2083, 31, 31, 32, 31, 31, 30, 30, 30, 29, 30, 30, 30],
    84: [2084, 31, 31, 32, 31, 31, 30, 30, 30, 29, 30, 30, 30],
    85: [2085, 31, 32, 31, 32, 30, 31, 30, 30, 29, 30, 30, 30],
    86: [2086, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 30, 30],
    87: [2087, 31, 31, 32, 31, 31, 31, 30, 30, 29, 30, 30, 30],
    88: [2088, 30, 31, 32, 32, 30, 31, 30, 30, 29, 30, 30, 30],
    89: [2089, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 30, 30],
    90: [2090, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 30, 30]
  };
  this.nepDigits = nepaliNumbers;
  this.engMonths = adMonths;
  this.nepaliMonthEnglish = bsMonthsEnglish;
  this.nepaliMonths = bsMonths;
  this.nepaliWeekdays = bsDaysFull;
  this.nepaliWeekdaysSmall = bsDays;
  this.engWeekDays = adDaysFull;
  this.nepDate = {};
  this.engDate = {};
  this.debugInfo = '';
};

var date = new BikramSambatConverter();
var Calender = function Calender(_ref) {
  var language = _ref.language,
      mode = _ref.mode,
      onDateClicked = _ref.onDateClicked,
      startDate = _ref.startDate,
      dayClickBehaviour = _ref.dayClickBehaviour;
  var tempDate = new Date(startDate);
  var nepaliDate = date.engToNepali(tempDate.getFullYear(), tempDate.getMonth() + 1, tempDate.getDate());
  var activeDate = nepaliDate;

  var _useState = useState(nepaliDate.year),
      year = _useState[0],
      setYear = _useState[1];

  var _useState2 = useState(nepaliDate.nmonth),
      month = _useState2[0],
      setMonth = _useState2[1];

  var _useState3 = useState([]),
      clickedDates = _useState3[0],
      setClickedDates = _useState3[1];

  var yearInfo = currentYear(year);
  var currMonth = yearInfo[month - 1];
  var years = getYears();

  var handleDateClick = function handleDateClick(year, month, date, day) {
    if (dayClickBehaviour === 'multiple') {
      var tempClickedDates = [].concat(clickedDates);
      var index = tempClickedDates.findIndex(function (clickedDate) {
        return clickedDate.year === year && clickedDate.month === month && clickedDate.day === day && clickedDate.date === date;
      });

      if (index !== -1) {
        tempClickedDates.splice(index, 1);
        setClickedDates(tempClickedDates);
        return onDateClicked(tempClickedDates);
      }

      tempClickedDates.push({
        year: year,
        month: month,
        date: date,
        day: day
      });
      setClickedDates(tempClickedDates);
      return onDateClicked(tempClickedDates);
    }

    setClickedDates([{
      year: year,
      month: month,
      date: date,
      day: day
    }]);
    return onDateClicked({
      year: year,
      month: month,
      date: date,
      day: day
    });
  };

  var handleMonthChange = function handleMonthChange(e) {
    clearDates();
    setMonth(e.target.value * 1);
  };

  var handleYearChange = function handleYearChange(e) {
    clearDates();
    setYear(e.target.value * 1);
  };

  var clearDates = function clearDates() {
    setClickedDates([]);
  };

  return /*#__PURE__*/React.createElement("main", {
    className: styles.calendarContain
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      textAlign: 'center'
    }
  }, mode === 'yearly' ? /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("select", {
    id: "year",
    value: year,
    onChange: handleYearChange
  }, years.map(function (year) {
    return /*#__PURE__*/React.createElement("option", {
      key: year,
      value: year
    }, language === 'nepali' ? date.toNepaliNumber(year) : year);
  })), /*#__PURE__*/React.createElement("select", {
    id: "month",
    value: month,
    onChange: handleMonthChange
  }, language === 'nepali' ? bsMonths.map(function (month, i) {
    return /*#__PURE__*/React.createElement("option", {
      key: month,
      value: i + 1
    }, month);
  }) : bsMonthsEnglish.map(function (month, i) {
    return /*#__PURE__*/React.createElement("option", {
      key: month,
      value: i + 1
    }, month);
  }))) : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("select", {
    id: "month",
    value: month,
    onChange: handleMonthChange
  }, language === 'nepali' ? bsMonths.map(function (month, i) {
    return /*#__PURE__*/React.createElement("option", {
      key: month,
      value: i + 1
    }, month);
  }) : bsMonthsEnglish.map(function (month, i) {
    return /*#__PURE__*/React.createElement("option", {
      key: month,
      value: i + 1
    }, month);
  })))), /*#__PURE__*/React.createElement("section", {
    className: styles.calendarDays
  }, /*#__PURE__*/React.createElement("section", {
    className: styles.calendarTopbar
  }, language === 'nepali' ? bsDays.map(function (day, i) {
    return /*#__PURE__*/React.createElement("span", {
      key: i,
      className: styles.topbarDays + " " + (i + 1 === 7 && styles.saturday)
    }, day);
  }) : adDays.map(function (day, i) {
    return /*#__PURE__*/React.createElement("span", {
      key: i,
      className: styles.topbarDays
    }, day);
  })), Array.from({
    length: Math.ceil((currMonth.weekStart + currMonth.endDate) / 7)
  }, function (_, i) {
    return i + 1;
  }).map(function (num) {
    return /*#__PURE__*/React.createElement("section", {
      key: num,
      className: styles.calendarWeek
    }, num === 1 && Array.apply(null, {
      length: currMonth.weekStart
    }).map(function (e, i) {
      return /*#__PURE__*/React.createElement("div", {
        key: i,
        className: styles.calendarDay + " " + styles.inactive
      }, /*#__PURE__*/React.createElement("span", {
        className: styles.calendarDate
      }, " "));
    }), Array.apply(null, {
      length: num === 1 ? 7 - currMonth.weekStart : num === Math.ceil((currMonth.weekStart + currMonth.endDate) / 7) ? 7 - (Math.ceil((currMonth.weekStart + currMonth.endDate) / 7) * 7 - currMonth.endDate - currMonth.weekStart) : 7
    }).map(function (e, i) {
      return /*#__PURE__*/React.createElement("div", {
        key: i,
        className: styles.calendarDay + " " + (clickedDates && clickedDates.every(function (date) {
          return date.date !== Math.max(0, 7 * (num - 1) - currMonth.weekStart) + (i + 1);
        }) ? null : styles.active) + " " + (activeDate.date === Math.max(0, 7 * (num - 1) - currMonth.weekStart) + (i + 1) && activeDate.nmonth === month && activeDate.year === year && styles.today) + " " + ((Math.max(0, 7 * (num - 1) - currMonth.weekStart) + (i + 1) + currMonth.weekStart) % 7 === 0 && styles.holiday),
        onClick: function onClick() {
          return handleDateClick(year, month, Math.max(0, 7 * (num - 1) - currMonth.weekStart) + (i + 1), (Math.max(0, 7 * (num - 1) - currMonth.weekStart) + (i + 1) + currMonth.weekStart) % 7 === 0 ? 7 : (Math.max(0, 7 * (num - 1) - currMonth.weekStart) + (i + 1) + currMonth.weekStart) % 7);
        }
      }, /*#__PURE__*/React.createElement("span", {
        className: styles.calendarDate
      }, language === 'nepali' ? date.toNepaliNumber(Math.max(0, 7 * (num - 1) - currMonth.weekStart) + (i + 1)) : Math.max(0, 7 * (num - 1) - currMonth.weekStart) + (i + 1)));
    }), num === Math.ceil((currMonth.weekStart + currMonth.endDate) / 7) && Array.apply(null, {
      length: Math.ceil((currMonth.weekStart + currMonth.endDate) / 7) * 7 - currMonth.endDate - currMonth.weekStart
    }).map(function (e, i) {
      return /*#__PURE__*/React.createElement("div", {
        key: i,
        className: styles.calendarDay + " " + styles.inactive
      }, /*#__PURE__*/React.createElement("span", {
        className: styles.calendarDate
      }, " "));
    }));
  })));
};
Calender.defaultProps = {
  language: 'english',
  mode: 'monthly',
  onDateClicked: function onDateClicked() {},
  dayClickBehaviour: 'single',
  startDate: new Date()
};
Calender.propTypes = {
  language: propTypes.string,
  mode: propTypes.string,
  onDateClicked: propTypes.func.isRequired,
  dayClickBehaviour: propTypes.string,
  startDate: propTypes.any
};
var useNepaliNUmber = function useNepaliNUmber(number) {
  return date.toNepaliNumber(number);
};

export { Calender, useNepaliNUmber };
